core/src/main/java/com/novoda/noplayer/exoplayer/InfoForwarder.java

```diff
@@ -0,0 +1,221 @@
+package com.novoda.noplayer.exoplayer;
+
+import android.view.Surface;
+
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.Format;
+import com.google.android.exoplayer2.decoder.DecoderCounters;
+import com.google.android.exoplayer2.upstream.DataSpec;
+import com.novoda.noplayer.listeners.InfoListeners;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+class InfoForwarder extends ExoPlayerEventForwarder {
+
+    private final InfoListeners infoListeners;
+
+    InfoForwarder(InfoListeners infoListeners) {
+        this.infoListeners = infoListeners;
+    }
+
+    @Override
+    public void onLoadStarted(DataSpec dataSpec, int dataType, int trackType, Format trackFormat, int trackSelectionReason,
```

Comment:
If you extract all these params in a class, then you can add logic into that object in order to return the hashmap (also you can test in isolation this new class) something like `exoPlayerLoadEvent.toMap()`